### ISSUE TITLE
Use a Group Managed service account to run a schedule task and make it "Run whether user is logged on or not"?

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#190]
These two ways are not working, here is my test code, I test it on windows server 2012R2:
```
using System;
using System.Runtime.InteropServices;
using Microsoft.Win32.TaskScheduler;

namespace WrapperDemo
{
    internal static class Program
    {
        public static void Main(string[] args)
        {
            try
            {
                var action = new ExecAction(@"C:\PowerSlim\test.cmd");
                var trigger = new DailyTrigger(1);
                trigger.StartBoundary = DateTime.Today.AddHours(20);
                TaskService.Instance.AddTask("msaTask", trigger, action, @"er\zz$", "", TaskLogonType.Password);
            }
            catch (COMException exception)
            {
                Console.WriteLine(exception.ToString());
            }

        }
    }
}

```
I got exception like following on the console output:

System.Runtime.InteropServices.COMException (0x8007052E): The username or password is incorrect. (Exception from HRESULT: 0x8007052E)
   at Microsoft.Win32.TaskScheduler.V2Interop.ITaskFolder.RegisterTaskDefinition(String Path, ITaskDefinition pDefinition, Int32 flags, Object UserId, Object password, TaskLogonType LogonType, Object sddl)
   at Microsoft.Win32.TaskScheduler.TaskFolder.RegisterTaskDefinition(String Path, TaskDefinition definition, TaskCreation createType, String UserId, String password, TaskLogonType LogonType, String sddl)
   at Microsoft.Win32.TaskScheduler.TaskService.AddTask(String path, Trigger trigger, Action action, String UserId, String Password, TaskLogonType LogonType)
   at WrapperDemo.Program.Main(String[] args) in C:\Users\lhuang2\Documents\Visual Studio 2015\Projects\WrapperDemo\WrapperDemo\Program.cs:line 20

Originally posted: 2016-03-30T08:13:26

**Comment by dahall** [comment]
Please try the assemblies in the attached zip file with the following code and let me know the results:
```
TaskService.Instance.AddTask("msaTask", QuickTriggerType.Daily,
   "C:\PowerSlim\test.cmd", null, "er\zz$", null, TaskLogonType.Password);
```

Originally posted: 2016-04-02T12:39:42

**Comment by dahall** [comment]
@dahall, I tried v4.5.2 in the attachment, it works.
But we need to work in .net framework 4.0, and we use NuGet to manage this library, so hope this fixing could be available soon. Thanks!

Originally posted: 2016-04-05T00:34:44

**Comment by dahall** [comment]
I got a communication from Microsoft saying to move public projects off .NET 4.0 and compile with 4.5.2 as it loads the 4.0 assemblies. I'll go back and include a 4.0 build in the NuGet package though in 2.5.17.

Originally posted: 2016-04-07T07:41:54

**Comment by dahall** [comment]
Fixed in 2.5.17 released 4/7/16

Originally posted: 2016-04-08T14:02:07
